client/SMCE_Client.cpp

```diff
@@ -1,339 +1,294 @@
+            const int pin = stoi(input.substr(3, 5));
+            const uint16_t value = stoi(input.substr(5));
+            auto apin = board_view.pins[pin].digital();
+            if (!apin.exists()) {
+                std::cout << "Pin does not exist!" << std::endl;
+            }else if (!apin.can_write()) {
+                std::cout << "Can't write to pin!" << std::endl;
+            }else{
+                if (value == 0) {
+                    apin.write(false);
+                } else if (value == 1) {
+                    apin.write(true);
+                } else {
+                    std::cout << "Value must be 0 or 1 for digital pins" << std::endl;
+                }
             }
 
-        }else if (input.starts_with("-ra ")){                                                                                    
-            bool read = true;                                                                                                    
-            int index_pin = stoi(input.substr(3));                                                                               
-            auto pin = board_view.pins[index_pin].analog();                                                                      
-            if(!pin.exists()){                                                                                                   
-                std::cout << "Pin does not exist!" << std::endl;                                                                 
-                read=false;                                                                                                      
-            }                                                                                                                    
-            if(!pin.can_read()){                                                                                                 
-                std::cout << "Can't read from pin!" << std::endl;                                                                
-                read=false;                                                                                                      
-            }                                                                                                                    
-            if(read){                                                                                                            
-                std::cout << "Value from pin " << index_pin << " is " << pin.read() << std::endl;                                
-            } 
+        } else if (input.starts_with("-ra ")) {
+            const int index_pin = stoi(input.substr(3));
+            auto pin = board_view.pins[index_pin].analog();
+            if (!pin.exists()) {
+                std::cout << "Pin does not exist!" << std::endl;
 
-        }else if (input.starts_with("-rd ")){
-            bool read = true;
-            int index_pin = stoi(input.substr(3));
+            }else if (!pin.can_read()) {
```

Comment:
Just because the board cannot read does not mean the user should not be able to see the value